test/tests/jobs/GetJobTest.cpp

```diff
@@ -0,0 +1,408 @@
+#include <test/lib/BedrockTester.h>
+
+struct GetJobTest : tpunit::TestFixture {
+    GetJobTest()
+        : tpunit::TestFixture("GetJob",
+                              BEFORE_CLASS(GetJobTest::setupClass),
+                              TEST(GetJobTest::getJob),
+                              TEST(GetJobTest::withNumResults),
+                              TEST(GetJobTest::noJobFound),
+                              TEST(GetJobTest::testPriorities),
+                              TEST(GetJobTest::testPrioritiesWithDifferentNextRunTimes),
+                              TEST(GetJobTest::testWithFinishedAndCancelledChildren),
+                              TEST(GetJobTest::testPrioritiesWithRunQueued),
+                              AFTER(GetJobTest::tearDown),
+                              AFTER_CLASS(GetJobTest::tearDownClass)) { }
+
+    BedrockTester* tester;
+
+    void setupClass() { tester = new BedrockTester({{"-plugins", "Jobs,DB"}}, {});}
+
+    // Reset the jobs table
+    void tearDown() {
+        SData command("Query");
+        command["query"] = "DELETE FROM jobs WHERE jobID > 0;";
+        tester->executeWaitVerifyContent(command);
+    }
+
+    void tearDownClass() { delete tester; }
+
+    // Simple GetJob
+    void getJob() {
+        // Create the job
+        SData command("CreateJob");
```

Comment:
You could DRY this file with a func to create a job that takes name, priority, nextRun, etc. Nab